packages/strapi-design-system/src/Select/SingleSelect.tsx

```diff
@@ -0,0 +1,330 @@
+                  {startIcon}
+                </Box>
+              )}
+              <Typography ellipsis textColor={value ? 'neutral800' : 'neutral600'}>
+                <RadixSelect.Value aria-label={value}>
+                  {value ? customizeContent(value) : placeholder}
+                </RadixSelect.Value>
+              </Typography>
+            </Flex>
+            <Flex as="span" gap={3}>
+              {value && onClear ? (
+                <IconBox
+                  role="button"
+                  tabIndex={1}
+                  onClick={handleClearClick}
+                  aria-disabled={disabled}
+                  aria-label={clearLabel}
+                  title={clearLabel}
+                  ref={clearRef}
+                >
+                  <Cross />
+                </IconBox>
+              ) : null}
+              <DownIcon>
+                <CarretDown />
+              </DownIcon>
+            </Flex>
+          </Trigger>
+          <RadixSelect.Portal>
+            <Content position="popper" sideOffset={4}>
+              <RadixSelect.Viewport ref={viewportRef}>
+                {children}
+                <Box id={intersectionId} width="100%" height="1px" />
+              </RadixSelect.Viewport>
+            </Content>
+          </RadixSelect.Portal>
+        </RadixSelect.Root>
+        <FieldHint />
+        <FieldError />
+      </Stack>
+    </Field>
+  );
+};
+
+interface TriggerProps {
+  $hasError: boolean;
+  $size: Required<Pick<SingleSelectProps, 'size'>>['size'];
+}
+
+const Trigger = styled(RadixSelect.Trigger)<TriggerProps>`
+  position: relative;
+  border: 1px solid ${({ theme, $hasError }) => ($hasError ? theme.colors.danger600 : theme.colors.neutral200)};
+  padding-right: ${({ theme }) => theme.spaces[3]};
+  padding-left: ${({ theme }) => theme.spaces[3]};
+  border-radius: ${({ theme }) => theme.borderRadius};
+  background: ${({ theme }) => theme.colors.neutral0};
+  overflow: hidden;
+  min-height: ${({ theme, $size }) => getThemeSize('input')({ theme, size: $size })};
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  gap: ${({ theme }) => theme.spaces[4]};
+
+  [aria-disabled='true'] {
+    color: ${(props) => props.theme.colors.neutral600};
+    background: ${(props) => props.theme.colors.neutral150};
+  }
+
+  ${({ theme, $hasError }) => inputFocusStyle()({ theme, hasError: $hasError })};
+`;
+
+const DownIcon = styled(RadixSelect.Icon)`
+  & > svg {
+    width: 6px;
+
+    & > path {
+      fill: ${({ theme }) => theme.colors.neutral600};
+    }
+  }
+`;
+
+const Content = styled(RadixSelect.Content)`
+  z-index: 4;
+  background: ${({ theme }) => theme.colors.neutral0};
+  box-shadow: ${({ theme }) => theme.shadows.filterShadow};
+  border: 1px solid ${({ theme }) => theme.colors.neutral150};
+  border-radius: ${({ theme }) => theme.borderRadius};
+  padding: ${({ theme }) => theme.spaces[1]};
+  width: var(--radix-select-trigger-width);
```

Comment:
Uff. Do you think we should hard-code this to avoid surprises?

Reply:
this is a dynamic value so hardcoding it would defeat the point of it, when the select is rendered it's width is recorded and then we can make the dropdown adhear to it because it renders far away – https://www.radix-ui.com/docs/primitives/components/select#constrain-the-content-size

Reply:
Are we sure radix will properly inject these properties into the parent application?

Reply:
I can't imagine why they wouldn't? If they don't, it's a bug in the library - I've not had any issue with radix.

We'll also be doing at least one beta incl. these changes anyway so we can revise if needs be.